src/services/TestResultsService.ts

```diff
@@ -501,4 +505,37 @@ export class TestResultsService {
     }
     return payload;
   }
+
+  private validateLecTestTypeFields(payload: ITestResultPayload) {
+    if (payload.testTypes) {
+      let missingField;
+      payload.testTypes.forEach((testType: { testTypeId: string; certificateNumber: string; expiryDate: Date; modType: any; emissionStandard: string; fuelType: string; testExpiryDate: any, testResult: string}) => {
+        if (this.isTestTypeLec(testType) ) {
+            if (!testType.certificateNumber) {
+              missingField = "Certificate number";
+              return;
+            }
+            if (testType.testResult === TEST_RESULT.PASS) {
+            if (!testType.testExpiryDate) {
+              missingField = "Expiry Date";
```

Comment:
Could potentially be missing multiple fields - better to return an array of missing fields?

Reply:
modified to return array.